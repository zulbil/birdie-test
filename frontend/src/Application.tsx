import * as React from "react";
import store, { webHistory } from "@App/store";
import { Provider } from "react-redux";
import LoginPage from "@App/components/LoginPage";
import Dashboard from "@App/components/Dashboard";

import { Route, Switch } from "react-router";
import { ConnectedRouter } from "connected-react-router";
import ProtectedRoute from "./components/ProtectedRoute";

export const Application = () => (
  <Provider store={store}>
    <ConnectedRouter history={webHistory}>
      <Switch>
        <Route exact={true} path="/" component={LoginPage} />
        <ProtectedRoute path="/dashboard" component={Dashboard} />
      </Switch>
    </ConnectedRouter>
  </Provider>
);
