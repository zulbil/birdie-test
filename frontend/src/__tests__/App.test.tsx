/// <reference types="jest" />

import * as React from "react";
import * as renderer from "react-test-renderer";
import { Application } from "../Application";

it("Renders correctly", () => {
  const tree = renderer.create(<Application />).toJSON();

  expect(tree).toMatchSnapshot();
});