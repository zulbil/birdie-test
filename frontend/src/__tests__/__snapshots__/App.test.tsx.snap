// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="sc-bdVaJa iSfbcZ"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root sc-bwzfXH drwwBR MuiPaper-rounded"
  >
    <h1
      className="sc-ifAKCX glJhFp"
    >
      Login to the service
    </h1>
    <h3
      className="sc-EHOje cEjmMk"
    >
      Use code we gave you to login to the service
    </h3>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
      >
        Code
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-85 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-86"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          cy-testId="codeField"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Code"
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <p
      className="sc-bxivhb glrLzi"
    >
      
    </p>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Sign in
      </span>
    </button>
  </div>
</div>
`;
